### PR TITLE
Feat/docker compose event

### PR DESCRIPTION
**Pull request by morhag90**
Close: #[issue-number]

## PR Details

Add new function and event createDockerComposeDevFile to support all configuration needed. 

### Summary
🆙🔌🐳

Added a new plugin event and functionality to generate a docker-compose file for the development environment of the server application. This allows plugins to customize the docker-compose configuration for different environments. Updated the `@amplication/code-gen-types` package version and type definitions accordingly.

> _`code-gen-types` changed_
> _New plugin event added_
> _Docker-compose for dev_

### Walkthrough
*  Update the version of the `@amplication/code-gen-types` package from 2.0.2 to 2.0.3 ([link](https://github.com/amplication/amplication/pull/6594/files?diff=unified&w=0#diff-d82899d07c853328e58d97565373ccdef6d66697428e75da289542f73d779ee7L3-R3), [link](https://github.com/amplication/amplication/pull/6594/files?diff=unified&w=0#diff-d82899d07c853328e58d97565373ccdef6d66697428e75da289542f73d779ee7L9-R9), [link](https://github.com/amplication/amplication/pull/6594/files?diff=unified&w=0#diff-848fe18cdea81eff2bb3d7e509a3d8e5a63a2974014e58ee366093f9fa9f2bf9L3-R3))
*  Add a new interface `CreateServerDockerComposeDevParams` to the `@amplication/code-gen-types` package to define the parameters for the new plugin event `CreateServerDockerComposeDev` ([link](https://github.com/amplication/amplication/pull/6594/files?diff=unified&w=0#diff-8ec1349510d73b28d516206efd7feaeaade54b394a928088f17e945bd9312ce3R132-R137), [link](https://github.com/amplication/amplication/pull/6594/files?diff=unified&w=0#diff-5c9fde1988b7fe0e213de34cd71c76a96dd419cb6ef73e332cb4f7d62b792a69R36))
*  Add a new plugin event name `CreateServerDockerComposeDev` to the `@amplication/code-gen-types` package to allow plugins to modify the content and properties of the docker-compose file for the development environment of the generated server ([link](https://github.com/amplication/amplication/pull/6594/files?diff=unified&w=0#diff-5c9fde1988b7fe0e213de34cd71c76a96dd419cb6ef73e332cb4f7d62b792a69R53), [link](https://github.com/amplication/amplication/pull/6594/files?diff=unified&w=0#diff-3811ed265e535dd8e3e3bd8618ca04a8bd0860e1cd5f0bd9e9cf2e8d321f321aR94))
*  Add a new constant `DOCKER_COMPOSE_DEV_FILE_NAME` to the `data-service-generator` package to define the name of the template file for the docker-compose file for the development environment of the generated server ([link](https://github.com/amplication/amplication/pull/6594/files?diff=unified&w=0#diff-2298f8ea5844c21b934d910ae47b0104c563658341d9e20cbbcb0b1e02cb29f8L14-R14))
*  Add a new function `createDockerComposeDevFile` to the `data-service-generator` package to create the docker-compose file for the development environment of the generated server by applying the plugin event `CreateServerDockerComposeDev` ([link](https://github.com/amplication/amplication/pull/6594/files?diff=unified&w=0#diff-9c1a5a055f94dfa5f9906d14a9dc82594d750eaa22bd3878db57d21e96f62aaaL20-R23), [link](https://github.com/amplication/amplication/pull/6594/files?diff=unified&w=0#diff-e956ce8f07f0653d9c6c9a8a71cae6466d9d9414fe168a39b69e9283b0eaf758R1-R88))
*  Call the `createDockerComposeDevFile` function inside the `createServerInternal` function in the `create-server.ts` file to add the docker-compose file for the development environment to the output of the server generation ([link](https://github.com/amplication/amplication/pull/6594/files?diff=unified&w=0#diff-9c1a5a055f94dfa5f9906d14a9dc82594d750eaa22bd3878db57d21e96f62aaaR111), [link](https://github.com/amplication/amplication/pull/6594/files?diff=unified&w=0#diff-9c1a5a055f94dfa5f9906d14a9dc82594d750eaa22bd3878db57d21e96f62aaaR130))
*  Rename the file `docker-compose.db.template.yml` to `docker-compose.dev.template.yml` in the `data-service-generator` package to reflect the new scope of the template file ([link](https://github.com/amplication/amplication/pull/6594/files?diff=unified&w=0#diff-245becbacee96e2bb9cc095f52c17a340cfbd16e896c5dac9b638292a67016d5))
*  Delete the file `create-docker-compose-db.ts` in the `data-service-generator` package as it is no longer needed ([link](https://github.com/amplication/amplication/pull/6594/files?diff=unified&w=0#diff-74089c34f74765395a7ac91debafe493f95f6883f7405e8e9b965900a9a1b6c1))

## PR Checklist

- [x] Tests for the changes have been added
- [x] `npm test` doesn't throw any error

### CODE REVIEW

File: packages/data-service-generator/src/server/docker-compose/create-docker-compose-dev.ts
Comment (overbit) [inline packages/data-service-generator/src/server/docker-compose/create-docker-compose-dev.ts:1]: @morhag90 how does this differs from the docker-compose file that we generate? 
Docker compose files are 99% of the time used only during development.

**Review by yuval-hazaz — CHANGES_REQUESTED** [review]

File: packages/data-service-generator/src/server/docker-compose/create-docker-compose-dev.ts
[the remark this answers is not in the document]
Comment (morhag90) [reply packages/data-service-generator/src/server/docker-compose/create-docker-compose-dev.ts:1]: @yuval-hazaz done

Comment (yuval-hazaz) [reply packages/data-service-generator/src/server/docker-compose/create-docker-compose-dev.ts:1]: @overbit Currently we generate two files:
docker-compose - with the DB, the actual service, and a migration container 
docker-compose-db - with just the DB 

This PR replaces the docker-compose-db and calls it docker-compose-dev so we can also add Kafka and other containers that may be needed without building the service

File: packages/data-service-generator/src/server/docker-compose/create-docker-compose-dev.ts [hunk 1]
```diff
@@ -9,15 +10,15 @@ import path from "path";
 import { prepareYamlFile } from "../../utils/prepare-yaml-file";
 
 import pluginWrapper from "../../plugin-wrapper";
-import { DOCKER_COMPOSE_DB_FILE_NAME } from "../constants";
+import { DOCKER_COMPOSE_DEV_FILE_NAME } from "../constants";
 import DsgContext from "../../dsg-context";
 
 export async function createDockerComposeDBFile(): Promise<ModuleMap> {
```
Comment (yuval-hazaz) [inline packages/data-service-generator/src/server/docker-compose/create-docker-compose-dev.ts:16]:
```suggestion
export async function createDockerComposeDevFile(): Promise<ModuleMap> {
```

**Review by yuval-hazaz — CHANGES_REQUESTED** [review]

File: packages/data-service-generator/src/server/docker-compose/create-docker-compose-dev.ts (hunk 1 above)
Comment (morhag90) [reply packages/data-service-generator/src/server/docker-compose/create-docker-compose-dev.ts]: @yuval-hazaz  Done

**Review by yuval-hazaz — APPROVED** [review]

**Pull request merged**